packages/shared/lib/utils/utils.ts

```diff
@@ -94,6 +94,22 @@ export function parseTokenExpirationDate(expirationDate: any): Date {
     return new Date(expirationDate);
 }
 
+export function parseTableauTokenExpirationDate(timeStr: string): Date | undefined {
+    // sample estimatedTimeToExpire: "estimatedTimeToExpiration": "177:05:38"
+    const [daysStr, hoursStr, minutesStr] = timeStr.split(':');
+
+    const days = Number(daysStr);
+    const hours = Number(hoursStr);
+    const minutes = Number(minutesStr);
+
+    if (isNaN(days) || isNaN(hours) || isNaN(minutes)) {
+        return;
+    }
+
+    const millisecondsToAdd = (days * 24 * 60 * 60 + hours * 60 * 60 + minutes * 60) * 1000;
+    return new Date(Date.now() + millisecondsToAdd);
```

Comment:
can be simplified slightly I think:
```
  const [days, hours, minutes] = timeStr.split(':').map(Number);
  
  if (days && hours && minutes) {
    const milliseconds = ((days * 24 + hours) * 60 + minutes) * 60 * 1000;
    return new Date(Date.now() + milliseconds);
  }
  
  return undefined;
```

Reply:
Ah, yes, that simplifies this function, updated in this latest [commit](https://github.com/NangoHQ/nango/pull/2526/commits/364d3113a7a21b0770c381f3a2702b7729d1fd86). Thank you! 🙏